### PR TITLE
Add types for terminal-kit

### PR DESCRIPTION
**Pull request by katsanva**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/cronvel/terminal-kit
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/terminal-kit/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  postfix?: string
+): string;
+
+export function stripEscapeSequences(str: string): string;
+
+export function stringWidth(str: string): number;
+
+export function truncateString(str: string, maxWidth: number): string;
+
+export { default as ScreenBufferHD } from "./ScreenBufferHD";
```
Comment (uniqueiniquity) [inline types/terminal-kit/index.d.ts]: Why are you exporting default here? I could be missing something here, but it doesn't look like these files have an `export default` to do that with.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: types/terminal-kit/index.d.ts (hunk 1 above)
Comment (katsanva) [reply types/terminal-kit/index.d.ts]: Thanks for pointing!

Comment (uniqueiniquity) [reply types/terminal-kit/index.d.ts]: Actually, I wouldn't have done it this way... 
It seems like the [underlying library](https://unpkg.com/terminal-kit@1.26.10/lib/ScreenBuffer.js) is not using `export default`, so these .d.ts files should not either. Instead, you should follow the appropriate template from [the TypeScript handbook](http://www.typescriptlang.org/docs/handbook/declaration-files/templates.html). Note that if you want to use `import name from 'package';` in your consuming code, you can use the `esModuleInterop` compiler setting.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

**Comment by katsanva** [comment]
@uniqueiniquity  Uhm... Sorry for bothering, but:

https://unpkg.com/terminal-kit@1.26.10/lib/Rect.js - related d.ts module already uses module-class.d.ts template
https://unpkg.com/terminal-kit@1.26.10/lib/ScreenBuffer.js - related d.ts module already uses module-class.d.ts template
https://unpkg.com/terminal-kit@1.26.10/lib/ScreenBufferHD.js - related d.ts module already uses module-class.d.ts template
https://unpkg.com/terminal-kit@1.26.10/lib/TextBuffer.js - related d.ts module already uses module-class.d.ts template
https://unpkg.com/terminal-kit@1.26.10/lib/Terminal.js - related d.ts module already uses module-class.d.ts template

This things give me a bunch of warnings like 
> The module "./Terminal" uses `export = `. Import with `import Terminal = require("./Terminal")`. See: https://github.com/Microsoft/dtslint/blob/master/docs/no-import-default-of-export-equals.md

so I need to use `requires`;

https://unpkg.com/terminal-kit@1.26.10/lib/termkit.js - related d.ts module already uses module.d.ts tepmlate

Can you please describe, how is it possible to avoid 

```
export { default as ScreenBufferHD } from "./ScreenBufferHD";
```

to export types from the under the hood modules right?

**Comment by uniqueiniquity** [comment]
@katsanva No bothering at all! This is what I'm here for. :)

The goal you should be aiming for is that people can use this package with out needing to have `esModuleInterop` enabled. That syntax does work for the way you're using it, but only because you have that flag enabled in this particular package; as it stands, anyone who includes this package in their own project will get errors without that flag.

Given that, I have a few recommendations:
- remove the line `export as namespace` from each of the files. As far as I can tell from the source, none of these files are actually adding something to the global namespace, and therefore need Node to run.
- Import and export the "under the hood" modules as follows:
```ts
import Rect = require("./Rect");
import ScreenBuffer = require("./ScreenBuffer");
import ScreenBufferHD = require("./ScreenBufferHD");
import Terminal = require("./Terminal");
import TextBuffer = require("./TextBuffer");

export { Rect, ScreenBuffer, ScreenBufferHD, Terminal, TextBuffer };
```
- You can still leave `esModuleInterop` enabled for this package, but only so you can continue to import in your test file as 
```ts
import t, {
  terminal as term,
  autoComplete as ac,
  getDetectedTerminal,
  ScreenBufferHD,
  ScreenBuffer,
  Terminal
} from "terminal-kit";
```
Please let me know if you have any more questions!

**Review by uniqueiniquity — APPROVED** [review]

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Pull request merged**